### PR TITLE
Replace Travis CI with Circle CI

### PR DESCRIPTION
**Pull request by kiequoo**
Due to the uncertain future of TravisCI (see https://medium.com/@dirk.avery/r-i-p-travis-ci-347753c73775).. let's switch to CirceCI

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    environment:
+      JVM_OPTS: -Xmx3200m
+      TERM: dumb
+
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - sbt-scala-cache-{{ checksum "build.sbt" }}
```
Comment (reinierl) [inline .circleci/config.yml:18]: clever. did you make this up yourself? :)

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+version: 2
+jobs:
```
Comment (reinierl) [inline .circleci/config.yml:2]: How would this work if we want to add builds for multiple JDKs as I did with Travis? Would we need multiple jobs? If so, is there a way to do it without massive duplication?

**Review by reinierl — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (kiequoo) [reply .circleci/config.yml:18]: Of course not :)

File: .circleci/config.yml (hunk 2 above)
Comment (kiequoo) [reply .circleci/config.yml:2]: I'm not sure TBH, this is the first time I use CirceCI

**Pull request merged**